File: eventweave/runtime/sinks/null.py
"""Null sink that counts events without writing them."""

from __future__ import annotations

from eventweave.core.event import Event
from eventweave.runtime.sink import Sink


class NullSink(Sink):
    """Count events without emitting them. Useful for dry-runs and tests."""

    def __init__(self) -> None:
        self._count = 0

    def open(self) -> None:
        pass

    def write(self, event: Event) -> None:
        self._count += 1

    def close(self) -> None:
        pass

    def flush(self) -> None:
        pass

    def count(self) -> int:
        return self._count

File: runtime-go/internal/sinks/null/null.go
package null

import "github.com/linxinhong/eventweave/runtime-go/internal/event"

// Sink counts events without writing them.
type Sink struct {
	count int
}

// New creates a null sink.
func New() *Sink { return &Sink{} }

// Open initializes the sink.
func (s *Sink) Open() error { return nil }

// Write increments the counter.
func (s *Sink) Write(ev event.Event) error {
	s.count++
	return nil
}

// Flush is a no-op.
func (s *Sink) Flush() error { return nil }

// Close is a no-op.
func (s *Sink) Close() error { return nil }

// Count returns counted events.
func (s *Sink) Count() int { return s.count }

// Failed always returns 0.
func (s *Sink) Failed() int { return 0 }
